server/src/main/java/org/opensearch/indices/replication/common/ReplicationLuceneIndex.java

```diff
@@ -121,6 +122,13 @@ public synchronized int totalFileCount() {
         return filesDetails.size();
     }
 
+    /**
+     * Returns true if set of files represent an incoming commit point
+     */
+    public synchronized boolean hasIncomingCommitPoint() {
+        return filesDetails.hasIncomingCommitPoint;
+    }
+
```

Comment:
Why does this have to be synchronised, since this is a read I guess? If this is for visibility can `volatile` suffice?

Reply:
Followed the convention of the other getter methods in this class. Thanks for the callout, I can make that change.